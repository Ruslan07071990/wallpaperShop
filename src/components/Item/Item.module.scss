.productCard {
  // max-width: 300px;
  margin-bottom: 40px;
  background-color: #fff;
  border-radius: 5px;
  box-shadow: 2px 4px 4px 0px rgba(0, 0, 0, 0.3);

  .imageWrapper {
    position: relative;
    width: 100%;
    // height: 200px;
    margin-bottom: 20px;
    display: flex;
    justify-content: center;
    overflow: hidden;
    border-radius: 5px;
    box-shadow: 2px 4px 4px 0px rgba(0, 0, 0, 0.3);

    // max-width: 320px;
    // min-height: 320px;
    background-size: cover;
  }

  & img {
    max-width: 320px;
    max-height: 320px;
    object-fit: cover;
  }

  .productName {
    height: 40px;
    margin: 0 15px 10px;
    font-size: 16px;
    text-align: center;
  }
  .count {
    // margin-left: 50px;
    font-size: 16px;
    font-weight: 600;
    color: blue;
    text-align: center;
  }

  .cardFooter {
    margin-bottom: 10px;
    display: flex;
    justify-content: space-between;
    align-items: end;

    .addFavorite {
      margin-bottom: 10px;
      margin-left: 15px;
      padding: 0;
      width: 30px;
      // height: 30px;
      display: flex;
      justify-content: center;
      align-items: center;
      border: none;
      border-radius: 50%;
      // border: 1.5px solid #fff;
      // backgroundColor: "#fff"
      // background-color: #fff;

      &:hover {
        background-color: #fff;
        border-radius: 60px;
      }

      &icon :hover {
        color: yellow;
      }
    }

    .productPrice {
      width: 100px;
      padding: 5px 10px;
      margin: 0 15px 10px;
      text-align: center;
      font-size: 16px;
      font-weight: 600;
      border-radius: 25px;
      color: #fff;
      background-color: #cc3333;
    }
  }

  .addtoCart {
    margin: 0 15px 15px;
    margin-left: 50px;
    width: 180px;
    // height: 30px;
    font-size: 16px;
    font-weight: 600;
    color: blue;
    border-radius: 25px;
    border: 1.5px solid #fff;
    

    &:hover {
      background-color: #dddddd;
      border-radius: 30px;
    }
  }

  .addRemoveBtnWrapper {
    margin-bottom: 15px;
    width: 180px;
    // height: 30px;
    padding: 0 15px;
    display: flex;
    justify-content: space-between;
    align-items: center;

    .minusFromCart,
    .plustoCart {
      width: 40px;
      height: 100%;
      margin: 0 15px;
      padding: 0;
      font-size: 24px;
    }

    .quantityProduct {
      width: 30px;
      height: 100%;
      color: blue;
    }
  }
}
.productList{
  width: 75vw;
  // height: 200px;
  box-sizing: border-box;
  padding: 12px 50px 12px 20px;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  gap: 20px;
  border-radius: 10px;
  border: 1px solid rgb(15, 64, 43);
  background-color: rgb(170, 123, 214);
  // position: relative;
  img{
    width: 200px;
    height: 300px;
  }
  .addtoCart {
    margin: 0 15px 15px;
    margin-left: 50px;
    width: 180px;
    // height: 30px;
    font-size: 16px;
    font-weight: 600;
    color: blue;
    border-radius: 25px;
    border: 1.5px solid #fff;
    

    &:hover {
      background-color: #dddddd;
      border-radius: 30px;
    }
  }
  .addFavorite {
    // margin-bottom: 10px;
    margin-left: 15px;
    padding: 0;
    width: 30px;
    height: 30px;
    display: flex;
    justify-content: center;
    align-items: center;
    border: none;
    border-radius: 50%;
   
   
    background-color: rgb(170, 123, 214);

 


  }
}
