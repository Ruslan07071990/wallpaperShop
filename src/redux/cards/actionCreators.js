import { SET_CARD } from "./actions";
import axios from 'axios';



export const setCards = (cards) => ({ type: SET_CARD, payload: cards });


export const axiosCards = () => {
    return async (dispatch) => {
        try {
            const { data } = await axios.get('./products.json');
            dispatch(setCards(data));

        } catch (error) {
            console.log('ERROR', error);
        }
    }
}
