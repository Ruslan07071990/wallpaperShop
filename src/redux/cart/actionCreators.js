import { SET_CART, ADD_TO_CART, DELETE_FROM_CART } from "./actions";
import { CART_LS_KEY } from "../../components/constants";
import { getStateFromLocalStorage } from "../../components/utils/localStorageHelper";

export const setCart = (cards) => ({
  type: SET_CART,
  payload: cards,
});

export const addToCart = (cards) => ({
  type: ADD_TO_CART,
  payload: cards,
});

export const deleteFromCart = (cards) => ({
  type: DELETE_FROM_CART,
  payload: cards,
});

export const axiosCart = () => {
  return async (dispatch) => {
    try {
      const cartFromLs = getStateFromLocalStorage(CART_LS_KEY);

      if (cartFromLs) {
        dispatch(setCart(cartFromLs));
      }

    } catch (error) {
      console.log("ERROR", error);
    }
  };
};
