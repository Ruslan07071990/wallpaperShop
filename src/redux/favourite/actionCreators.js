import { SET_FAVOURITE, CHANGE_FAVOURITE } from "./actions";
import { FAV_LS_KEY } from "../../components/constants";
import { getStateFromLocalStorage } from "../../components/utils/localStorageHelper";

export const setFavourite = (cards) => ({
  type: SET_FAVOURITE,
  payload: cards,
});

export const changeFavourite = (cards) => ({
  type: CHANGE_FAVOURITE,
  payload: cards,
});

export const axiosFavourite = () => {
  return async (dispatch) => {
    try {
      const favFromLs = getStateFromLocalStorage(FAV_LS_KEY);

      if (favFromLs) {
        dispatch(setFavourite(favFromLs));
      }

    } catch (error) {
      console.log("ERROR", error);
    }
  };
};
