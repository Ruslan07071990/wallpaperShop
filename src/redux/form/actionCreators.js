import { FORM_CHEC } from "./actions";



export const formBuyChec = (value) => ({
    type: FORM_CHEC,
    payload: value,
  });