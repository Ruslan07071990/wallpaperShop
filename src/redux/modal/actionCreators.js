import { SET_MODAL, SET_MODAL_ACTION, SET_MODAL_TITLE,PRODUKT_PRICE,PRODUKT_NAME } from "./actions";

export const setModal = () => ({ type: SET_MODAL });

export const setModalAction = (value) => ({
  type: SET_MODAL_ACTION,
  payload: value,
});
export const setModalTitle = (value) => ({
  type: SET_MODAL_TITLE,
  payload: value,
});

export const setProductName = (value) => ({
  type: PRODUKT_NAME,
  payload: value,
});
export const setProductPrice = (value) => ({
  type: PRODUKT_PRICE,
  payload: value,
});

