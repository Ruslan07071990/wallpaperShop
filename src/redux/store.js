import {createStore, applyMiddleware} from "redux"
import appReduser from "./appRedusers";
import { composeWithDevTools } from "redux-devtools-extension";
import thunk from "redux-thunk";

const store = createStore(appReduser,composeWithDevTools(applyMiddleware(thunk)));
export default store;