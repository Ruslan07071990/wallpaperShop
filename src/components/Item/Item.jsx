import React from "react";
import { changeFavourite } from "../../redux/favourite/actionCreators";
import { ReactComponent as StarIcon } from "./svg/StarIcon.svg";
import { ReactComponent as StarIconOutline } from "./svg/StarIconOutline.svg";
import styles from "./Item.module.scss";
import { useDispatch, useSelector } from "react-redux";
import PropTypes from "prop-types";
import Button from "../Button/Button";
import { addToCart } from "../../redux/cart/actionCreators";
import {
  setModal,
  setModalAction,
  setModalTitle,
  setProductName,
  setProductPrice,
} from "../../redux/modal/actionCreators";
import classNames from "classnames";
import ViewContext from "../../context";
import { useContext } from "react";

const Item = (props) => {
  const { id, name, price, image, count } = props;
  const { view } = useContext(ViewContext);
  const dispatch = useDispatch();
  const toggleModal = () => {
    dispatch(setModal());
  };
  const handleAddToCart = () => {
    dispatch(
      setModalAction(
        <Button
          onClick={() => {
            dispatch(addToCart({ name, price, image, id }));
            toggleModal();
          }}
          text="Додати в кошик"
        />
      )
    );
    dispatch(setModalTitle("Додати товар в Кошик?"));

    dispatch(setProductName(name));
    dispatch(setProductPrice(price));
    toggleModal();
  };
  const changeFavourites = () => {
    dispatch(changeFavourite({ name, price, image, id }));
  };
  const favourites = useSelector((state) => state.favourite.cards);
  const isFavorited = !!favourites.find((el) => el.id === id);

  return (
    <>

      <div 
      className={classNames(styles.productCard,{
        [styles.productCard]: view === "viewCards",
        [styles.productList]: view === "viewList",
      })}>
        <img className={styles.imageWrapper} src={`./img/${image}`} alt="" />

        <h3 className={styles.productName}>{name}</h3>
        <p className={styles.count}> Кількість {count}</p>

        <div className={styles.cardFooter}>
          <button className={styles.addFavorite} onClick={changeFavourites}>
            {isFavorited ? <StarIconOutline /> : <StarIcon data-testid ="starIcon" />}
          </button>

          <p className={styles.productPrice}>{price} грн</p>
        </div>

        <button className={styles.addtoCart} onClick={handleAddToCart}>
          <span>Додати до Кошика</span>
        </button>
      </div>
    </>
  );
};

Item.propTypes = {
  name: PropTypes.string,
  price: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  image: PropTypes.string,
  count: PropTypes.number,
  id: PropTypes.number,
};

Item.defaultProps = {
  name: "",
  price: "",
  image: "",
  count: '',
  id: '',
};

export default Item;
