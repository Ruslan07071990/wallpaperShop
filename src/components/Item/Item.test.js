import Item from "./Item";
import { render } from "@testing-library/react";
import { Provider, useDispatch } from "react-redux";
import { store } from "../../redux/store";
import { useEffect } from "react";
import { setModal } from "../../redux/modal/actionCreators";
import isFavorited from "./Item";

// MyComponent.test.js
import React from "react";

import { useSelector as useSelectorMock } from "react-redux";

// Mock useSelector
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));

const yourObject = {
  find: jest.fn().mockReturnThis(),
};

yourObject.find.mockImplementation(function () {
  return false;
});
// {
//     id: 3,
//     name: "Ford mustang",
//     price: 60900,
//     image: "3.jpg",
//     article: "1-3",
//     color: "Помаранчевий",
//     iSFavorites: false,
//   };
// Mock useContext
// jest.mock('./MyContext', () => ({
//   MyContext: {
//     Consumer: ({ children }) => children('MockContextValue'),
//   },
// }));

describe("MyComponent", () => {
  it("renders selected value and context value", () => {
    // Mock the useSelector implementation
    useSelectorMock.mockReturnValue(yourObject);

    const { getByText, getByTestId } = render(<Item />);

    expect(getByTestId("starIcon")).toBeInTheDocument();
  });
  it("should Item render", () => {
    useSelectorMock.mockReturnValue(yourObject);
    const { asFragment } = render(<Item name="Ford mustang" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
