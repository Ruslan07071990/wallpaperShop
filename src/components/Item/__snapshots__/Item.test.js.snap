// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should Item render 1`] = `
<DocumentFragment>
  <div
    class="productCard"
  >
    <img
      alt=""
      class="imageWrapper"
      src="./img/"
    />
    <h3
      class="productName"
    >
      Ford mustang
    </h3>
    <p
      class="count"
    >
       Кількість 
    </p>
    <div
      class="cardFooter"
    >
      <button
        class="addFavorite"
      >
        <svg
          data-testid="starIcon"
        >
          StarIcon.svg
        </svg>
      </button>
      <p
        class="productPrice"
      >
         грн
      </p>
    </div>
    <button
      class="addtoCart"
    >
      <span>
        Додати до Кошика
      </span>
    </button>
  </div>
</DocumentFragment>
`;
